File: examples/server_cgi_on_request.pl
use Mojo::Server::CGI;

my $cgi = Mojo::Server::CGI->new;
$cgi->unsubscribe('request');
$cgi->on(request => sub {
    my ($cgi, $tx) = @_;

    # Request
    my $method = $tx->req->method;
    my $path   = $tx->req->url->path;

    # Response
    $tx->res->code(200);
    $tx->res->headers->content_type('text/plain');
    $tx->res->body("$method request for $path\n");

    # Resume transaction
    $tx->resume;
});
$cgi->run;

File: examples/server_cgi_on_request.py
import Pyjo.Server.CGI
from Pyjo.Util import b

cgi = Pyjo.Server.CGI.new()
cgi.unsubscribe('request')


@cgi.on
def request(cgi, tx):
    # Request
    method = tx.req.method
    path = tx.req.url.path

    # Response
    tx.res.code = 200
    tx.res.headers.content_type = 'text/plain'
    tx.res.body = b("{0} request for {1}\n".format(method, path))

    # Resume transaction
    tx.resume()


cgi.run()
